### ISSUE TITLE
Create topics sync returns before the topics are actually created

### ISSUE THREAD

**Issue by kobik** [SOHU-Co/kafka-node#622]
Hi,

I use the following code in order to create a topic
```javascript
function createTopics(topics) {
    console.log('Creating kafka topics', topics);
    var promise = new Promise(function (resolve, reject) {
        producer.createTopics(topics, false, function (err, res) {
            if (err) {
                console.error('Failed to create topic(s)', err);
                return reject(err);
            }
            console.log('Kafka topic(s) created');
            return resolve(res);
        });
    });
    return promise;
}
```

The output looks like this:
```
Creating kafka topics [ 'test3' ]
  kafka-node:Client create topics currentAttempt 1 +0ms
Kafka topic(s) created
  kafka-node:Client kafka reported topics [] +114ms
  kafka-node:Client Topics left test3 +15ms
  kafka-node:Client create topics currentAttempt 2 +204ms
  kafka-node:Client kafka reported topics [ 'test3' ] +8ms
  kafka-node:Client Topics created test3 +1ms
```

As you can see, "Kafka topic(s) created" message is printed before the topic was actually created.
Is this a bug or the expected behavior?

Thanks

**Comment by hyperlink** [comment]
I see the wording in the readme is confusing about this but passing `false` should yield the callback right away.

**Comment by kobik** [comment]
So how can I wait or verify that the topics were created?

**Comment by hyperlink** [comment]
If you pass in true instead the callback will be called once it has verified all topics were created (or it ran out of retries).

**Comment by kobik** [comment]
Thanks

**Comment by CharlesWall** [comment]
I think you updated this already right? can this be closed?

**Comment by 0x530302** [comment]
Not quite, the API in the Readme.md should be updated, since the wording is very confusing.

I suggest something like

> When async is set to `true`, the callback is called after it has been verified that all topics were created or an error ocurred. If async is set to `false`, the callback will be called immediately without verifying the creation.

instead of

> When async is set to `false`, this method does not return until all topics are created, otherwise it returns immediately.

At least that is how I interpret the source code of the createTopics method ;)

In addition, I think it's very confusing that the method throws an exception when one of the topics has an illegal name, only if async is set to `false` and the callback is called with that error if it's set to `true`.
